resqpy/well/_wellbore_marker_frame.py

```diff
@@ -115,130 +108,97 @@ def from_dataframe(cls,
+
+        return wellbore_marker_frame
 
     @staticmethod
-    def __get_interp_type(boundary_kind):
-        """Return the interpretation object's type based on the boundary kind.
+    def __parse_dataframe_row_for_interp_uuid(parent_model, row, row_marker_type, interp_citation_title_col):
+        """Extract the uuid of the interpretation object that the marker relates to from a row in the dataframe."""
 
-        arguments:
-            boundary_kind (string): valid kind of boundary feature
-            Valid kinds include "fault", "geobody", "boundary" and "horizon" # TODO: add "fracture as valid type
-
-        returns:
-           string
-        """
-
-        if boundary_kind == 'horizon':
-            interp_type = 'obj_HorizonInterpretation'
-        elif boundary_kind == 'fault':
-            interp_type = 'obj_FaultInterpretation'
-        elif boundary_kind == 'geobody':
-            interp_type = 'obj_GeobodyInterpretation'
+        if row_marker_type in ['horizon', 'fault', 'geobody']:
+            row_interp_type = 'obj_' + row_marker_type.capitalize() + 'Interpretation'
         else:
-            interp_type = None
-        return interp_type
-
-    def get_trajectory_obj(self, trajectory_uuid):
-        """Returns a trajectory object.
-
-        arguments:
-           trajectory_uuid (string or uuid.UUID): the uuid of the trajectory for which a Trajectory object is required
-
-        returns:
-           well.Trajectory object
-
-        note:
-           this method is not usually called directly
-        """
+            row_interp_type = None
+        try:
+            if row_interp_type is not None:
+                row_interp_citation_title = row[interp_citation_title_col]
+                row_interp_part = parent_model.part(obj_type = row_interp_type, title = row_interp_citation_title)
+                row_interp_uuid = parent_model.uuid_for_part(part_name = row_interp_part)
+                assert row_interp_uuid is not None, 'interpretation uuid cannot be found'
+            else:
+                # no interpretation exists for the boundary feature
+                row_interp_uuid = None
+        except KeyError:  # no interpretation column in the dataframe
+            row_interp_uuid = None
 
-        if trajectory_uuid is None:
-            log.error('no trajectory was found')
-            return None
-        else:
-            # create new trajectory object
-            trajectory_root_node = self.model.root_for_uuid(trajectory_uuid)
-            assert trajectory_root_node is not None, 'referenced wellbore trajectory missing from model'
-            return Trajectory(self.model, uuid = trajectory_uuid)
+        return row_interp_uuid
 
-    def get_interpretation_obj(self, interpretation_uuid, interp_type = None):
-        """Creates an interpretation object; returns a horizon or fault interpretation object.
+    # TODO: discuss making this a regular method as the markers require a parent wellbore marker frame
```

Comment:
Yes, let's talk this over again: perhaps we want to be able to instantiate WellboreMarker objects with None as the parent marker frame, then this (as a class method) would fill in that attribute.  But that might be overkill.

Reply:
As it stands, the function doesn't really deliver much. I think we either scrap it, adopt what I wrote just above, or modify it to be class method that creates a marker frame with a subset of markers from an existing frame in some way.  Let's discuss.